extensions/oidc/runtime/src/main/java/io/quarkus/oidc/OidcTenantConfig.java

```diff
@@ -3024,14 +3027,19 @@ public static OidcTenantConfigBuilder builder(io.quarkus.oidc.runtime.OidcTenant
     }
 
     /**
-     * Creates {@link OidcTenantConfig} from the {@code mapping}. This method is more efficient than
-     * the {@link #builder()} method if you don't need to modify the {@code mapping}.
+     * Creates {@link OidcTenantConfig} from the {@code mapping}.
      *
      * @param mapping tenant config as returned from the SmallRye Config; must not be null
      * @return OidcTenantConfig
      */
     public static OidcTenantConfig of(io.quarkus.oidc.runtime.OidcTenantConfig mapping) {
-        return new OidcTenantConfig(mapping);
+        if (mapping == null) {
```

Comment:
Hi Michal, can you clarify what is happening here ?

Reply:
`OidcTenantConfig` can be produced by 3 things:

1. SmallRye Config
2. user with `OidcTenantConfigBuilder`
3. user with legacy `OidcTenantConfig` constructor

Quarkus OIDC has 2 entrypoints how it consumes `OidcTenantConfig`:

- `OidcImpl`
- `TenantConfigResolver`

both of these entrypoints can receive config created either by `1.`, `2.` or `3.`.

The `OidcTenantConfig` used by Quarkus OIDC can never even see `TenantContextFactory` where we previously set _dynamic_ default values. This method you are commenting on guarantee correct default values in all situations where Quarkus OIDC extension is using `OidcTenantConfig`.

Reply:
@michalvavrik Can you please give an example of what can happen without this code, please give a sequence

Reply:
This function makes deep copy of original config and applies defaults previously applied in `TenantContextFactory`.

1. You keep modifying original instance and you use fields and setters marked for removal, this `of` method you are commenting to applies builder if defaults are not correct
2. User can change original field on a different thread (different incoming request) and this may or may not effect different thread (different incoming request). `io.quarkus.oidc.TenantConfigResolver#resolve` has `RoutingContext` and result is request-specific. However Quarkus OIDC treats result as if `io.quarkus.oidc.TenantConfigResolver#resolve` was mathematical function that returns always same value.